--- linux-4.4.268/drivers/char/lrng/lrng_drbg.c.orig	2021-05-14 13:11:07.327330888 -0400
+++ linux-4.4.268/drivers/char/lrng/lrng_drbg.c	2021-05-14 13:13:44.117414815 -0400
@@ -108,6 +108,10 @@ static void *lrng_drbg_drng_alloc(u32 se
 	if (ret)
 		goto err;
 
+	ret = drbg->d_ops->crypto_init(drbg);
+	if (ret)
+		goto dealloc;
+
 	if (sec_strength > drbg_sec_strength(drbg->core->flags)) {
 		pr_err("Security strength of DRBG (%u bits) lower than requested by LRNG (%u bits)\n",
 			drbg_sec_strength(drbg->core->flags) * 8,
