--- linux-5.4.203/include/crypto/sha.h.orig	2022-08-01 20:37:46.742512629 +0200
+++ linux-5.4.203/include/crypto/sha.h	2022-08-01 20:41:09.317200142 +0200
@@ -159,4 +159,14 @@ extern int sha224_update(struct sha256_s
 			 unsigned int length);
 extern int sha224_final(struct sha256_state *sctx, u8 *hash);
 
+/*
+ * An implementation of SHA-1's compression function.  Don't use in new code!
+ * You shouldn't be using SHA-1, and even if you *have* to use SHA-1, this isn't
+ * the correct way to hash something with SHA-1 (use crypto_shash instead).
+ */
+#define SHA1_DIGEST_WORDS	(SHA1_DIGEST_SIZE / 4)
+#define SHA1_WORKSPACE_WORDS	16
+void sha_init(__u32 *buf);
+void sha_transform(__u32 *digest, const char *data, __u32 *W);
+
 #endif
