--- linux-4.9.321/kernel/sched/core.c.orig	2022-07-11 07:43:38.448730624 +0200
+++ linux-4.9.321/kernel/sched/core.c	2022-07-11 07:43:57.633884127 +0200
@@ -25,6 +25,9 @@
  *  2007-11-29  RT balancing improvements by Steven Rostedt, Gregory Haskins,
  *              Thomas Gleixner, Mike Kravetz
  */
+#include <linux/highmem.h>
+#include <linux/hrtimer_api.h>
+#include <linux/ktime_api.h>
 
 #include <linux/kasan.h>
 #include <linux/mm.h>
