--- linux-4.14/drivers/char/lrng/lrng_proc.c.orig	2020-08-21 17:08:25.660195920 +0200
+++ linux-4.14/drivers/char/lrng/lrng_proc.c	2020-08-21 17:27:39.738051426 +0200
@@ -154,9 +154,22 @@ static int lrng_proc_type_show(struct se
 	return 0;
 }
 
+static int lrng_proc_type_open(struct inode *inode, struct file *file)
+{
+	return single_open(file, lrng_proc_type_show, NULL);
+}
+
+static const struct file_operations lrng_proc_ops = {
+	.owner		= THIS_MODULE,
+	.open		= lrng_proc_type_open,
+	.read		= seq_read,
+	.llseek		= seq_lseek,
+	.release	= single_release,
+};
+
 static int __init lrng_proc_type_init(void)
 {
-	proc_create_single("lrng_type", 0444, NULL, &lrng_proc_type_show);
+	proc_create("lrng_type", 0444, NULL, &lrng_proc_ops);
 	return 0;
 }
 
