--- linux-5.4/drivers/char/lrng/lrng_sw_noise.c.orig	2020-10-10 21:00:01.282688553 +0200
+++ linux-5.4/drivers/char/lrng/lrng_sw_noise.c	2020-10-10 21:12:07.975025774 +0200
@@ -12,6 +12,7 @@
 #include <crypto/hash.h>
 #include <linux/lrng.h>
 #include <linux/random.h>
+#include <linux/ratelimit.h>
 
 #include "lrng_internal.h"
 #include "lrng_sw_noise.h"
@@ -377,7 +378,7 @@ static inline void lrng_time_process(u32
 	if (unlikely(!lrng_state_fully_seeded())) {
 
 		/* Seed random32 PRNG with data not used by LRNG. */
-		this_cpu_add(net_rand_state.s1, random32_data);
+		//this_cpu_add(net_rand_state.s1, random32_data);
 
 		if (lrng_raw_hires_entropy_store(now_time))
 			goto out;
@@ -396,8 +397,8 @@ static inline void lrng_time_process(u32
 			goto out;
 
 		/* Seed random32 PRNG with data not used by LRNG. */
-		this_cpu_add(net_rand_state.s1,
-			(now_time & ~LRNG_DATA_SLOTSIZE_MASK) ^ random32_data);
+		//this_cpu_add(net_rand_state.s1,
+		//	(now_time & ~LRNG_DATA_SLOTSIZE_MASK) ^ random32_data);
 
 		health_test = lrng_health_test(now_time_masked);
 		if (health_test > lrng_health_fail_use)
