--- linux-4.19/drivers/char/lrng/lrng_sha1.c.orig	2022-05-17 16:15:25.074502863 -0400
+++ linux-4.19/drivers/char/lrng/lrng_sha1.c	2022-05-17 16:15:38.670547955 -0400
@@ -30,7 +30,7 @@ static void lrng_sha1_block_fn(struct sh
 	u32 temp[SHA1_WORKSPACE_WORDS];
 
 	while (blocks--) {
-		sha1_transform(sctx->state, src, temp);
+		sha_transform(sctx->state, src, temp);
 		src += SHA1_BLOCK_SIZE;
 	}
 	memzero_explicit(temp, sizeof(temp));
