--- linux-5.4/drivers/char/lrng/lrng_chacha20.c.orig	2020-08-20 13:17:09.072202669 +0200
+++ linux-5.4/drivers/char/lrng/lrng_chacha20.c	2020-08-20 13:18:16.877016437 +0200
@@ -172,14 +172,14 @@ void __init lrng_cc20_init_state_boot(st
 	u32 i;
 
 	for (i = 0; i < CHACHA_KEY_SIZE_WORDS; i++) {
-		if (arch_get_random_seed_long_early(&v) ||
-		    arch_get_random_long_early(&v))
+		if (arch_get_random_seed_long(&v) ||
+		    arch_get_random_long(&v))
 			chacha20->key.u[i] ^= v;
 	}
 
 	for (i = 0; i < 3; i++) {
-		if (arch_get_random_seed_long_early(&v) ||
-		    arch_get_random_long_early(&v))
+		if (arch_get_random_seed_long(&v) ||
+		    arch_get_random_long(&v))
 			chacha20->nonce[i] ^= v;
 	}
 
--- linux-5.4/drivers/char/lrng/lrng_pool.c.orig	2020-08-20 13:19:33.143931761 +0200
+++ linux-5.4/drivers/char/lrng/lrng_pool.c	2020-08-20 13:19:49.044122592 +0200
@@ -578,8 +578,8 @@ int __init rand_initialize(void)
 
 	lrng_pool_lfsr_u32(now_time);
 	for (i = 0; i < LRNG_POOL_SIZE; i++) {
-		if (!arch_get_random_seed_long_early(&rand) &&
-		    !arch_get_random_long_early(&rand))
+		if (!arch_get_random_seed_long(&rand) &&
+		    !arch_get_random_long(&rand))
 			rand = random_get_entropy();
 		lrng_pool_lfsr((u8 *)&rand, sizeof(rand));
 	}
