--- linux-4.19/include/crypto/hash.h.orig	2020-10-15 20:26:21.099078090 +0200
+++ linux-4.19/include/crypto/hash.h	2020-10-15 20:27:02.427257809 +0200
@@ -151,9 +151,11 @@ struct shash_desc {
 	void *__ctx[] CRYPTO_MINALIGN_ATTR;
 };
 
+#define HASH_MAX_DESCSIZE	(sizeof(struct shash_desc) + 360)
+
 #define SHASH_DESC_ON_STACK(shash, ctx)				  \
 	char __##shash##_desc[sizeof(struct shash_desc) +	  \
-		crypto_shash_descsize(ctx)] CRYPTO_MINALIGN_ATTR; \
+		HASH_MAX_DESCSIZE] CRYPTO_MINALIGN_ATTR; \
 	struct shash_desc *shash = (struct shash_desc *)__##shash##_desc
 
 /**
